### DISCUSSION TITLE
Swagger Replacement in ASP.NET 9

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#118] [General]
# Swagger Replacement in ASP.NET 9

You might have heard that Swagger is removed in the templates of ASP.NET 9. So this blog post will show you alternatives.

https://steven-giesel.com/blogPost/777776a2-a13c-46bb-bf6b-89316d88508c/swagger-replacement-in-aspnet-9

**Comment by Patrick-3000** [comment]
In my opinion, these two statements contradict each other:

> Mainly because the team wants to leverage their own implementation and don't want to depend on Swashbuckle, which, according to the thread, is not longer maintained.

and

> You can just install Swashbuckle again. It's still maintained and works perfectly.

Could you please clarify this?

**Reply by linkdotnet** [reply]
The .NET Team itself deemed Swashbuckle isn't actively maintained anymore. When they made the announcement, there weren't many commits.

But that doesn't mean you can't use it. For local development, there is no reason not to stick to that even though .NET itself dropped it in its templates. And (maybe because of that announcement) the Swashbuckle team did release some new versions.

Therefore, it might be perfectly fine to continue using it or even if you start from a .NET 9 template to install the dependency on your own.

**Reply by Patrick-3000** [reply]
Thank you!
